calculator.go

```diff
@@ -2,90 +2,109 @@
 package calculator
 
 import (
-	"errors"
+	"fmt"
 	"math"
 )
 
-var (
-	// errNotEnoughNumbers is returned by functions that expects
-	// minimum of two arguments.
-	errNotEnoughNumbers = errors.New("minimum two numbers are required")
+// Add takes two or more numbers and returns the result of adding them together.
+func Add(a, b float64, c ...float64) float64 {
+	result := a + b
 
-	errDivisionByZero     = errors.New("division by zero")
-	errSqrtNumberNegative = errors.New("sqrt from negative number")
-)
-
-// Add takes 2 or more numbers and returns the result of adding them together.
-// If the number of arguments is less than two the function will return an error.
-func Add(a ...float64) (float64, error) {
-	if len(a) < 2 {
-		return 0, errNotEnoughNumbers
+	for _, i := range c {
+		result += i
 	}
-	var result float64
 
-	for _, i := range a {
-		result = result + i
-	}
-
-	return result, nil
+	return result
 }
 
 // Subtract takes two or more numbers and returns the result
-// of subtracting them or error, if the number of arguments
-// is less than two.
-func Subtract(a ...float64) (float64, error) {
-	if len(a) < 2 {
-		return 0, errNotEnoughNumbers
-	}
-	result := a[0]
+// of subtracting them.
+func Subtract(a, b float64, c ...float64) float64 {
+	result := a - b
 
-	for i := 1; i < len(a); i++ {
-		result = result - a[i]
+	for _, i := range c {
+		result -= i
 	}
 
-	return result, nil
+	return result
 }
 
-// Multiply takes min two numbers and returns the product
-// or error, if the number of arguments is less than two.
-func Multiply(a ...float64) (float64, error) {
-	if len(a) < 2 {
-		return 0, errNotEnoughNumbers
-	}
-	result := a[0]
+// Multiply takes two or more numbers and returns the product.
+func Multiply(a, b float64, c ...float64) float64 {
+	result := a * b
 
-	for i := 1; i < len(a); i++ {
-		result = result * a[i]
+	for _, i := range c {
+		if i == 0 {
+			return 0
+		}
+		result *= i
 	}
 
-	return result, nil
+	return result
 }
 
-// Divide takes n numbers and return the result of division.
-// It returns error if the n+1 argument is equal 0 (division by zero)
-// or if the numer of arguments is less than 2.
-func Divide(a ...float64) (float64, error) {
-	if len(a) < 2 {
-		return 0, errNotEnoughNumbers
+// Divide takes two or more numbers and return the result of division.
+// It returns division by zero error if the n+1 argument is equal 0.
+func Divide(a, b float64, c ...float64) (float64, error) {
+	if a == 0 {
```

Comment:
Another special case that, technically, is not required. In general the simplest (correct) program is the best, so I would avoid adding performance-oriented short cuts until there's a genuine need to do so—because they're not free. We pay the cost in terms of simplicity and readability.